dashboard/client/src/type/serve.ts

```diff
@@ -0,0 +1,64 @@
+export enum ServeApplicationStatus {
+  // Keep in sync with ApplicationStatus in python/ray/serve/_private/common.py
+  NOT_STARTED = "NOT_STARTED",
+  DEPLOYING = "DEPLOYING",
+  RUNNING = "RUNNING",
+  DEPLOY_FAILED = "DEPLOY_FAILED",
+  DELETING = "DELETING",
+}
+
+export type ServeApplication = {
+  name: string;
+  route_prefix: string;
+  docs_path: string | null;
+  status: ServeApplicationStatus;
+  message: string;
+  last_deployed_time_s: number;
+  deployed_app_config: Record<string, any>;
+  deployments: {
+    [name: string]: ServeDeployment;
+  };
+};
+
+export enum ServeDeploymentStatus {
+  // Keep in sync with DeploymentStatus in python/ray/serve/_private/common.py
+  UPDATING = "UPDATING",
+  HEALTHY = "HEALTHY",
+  UNHEALTHY = "UNHEALTHY",
+}
+
+export type ServeDeployment = {
+  name: string;
+  status: ServeDeploymentStatus;
+  message: string;
+  deployment_config: ServeDeploymentConfig;
+};
+
+export type ServeDeploymentConfig = {
```

Comment:
Is there any way to proactively keep these configs in sync with the Serve schemas? Is there any way to write unit tests in TypeScript or Python that would raise an error if they went out of sync?

Reply:
We want to generate typescript types from the python code in the future. I will leave this for now until we do so.

Reply:
Sounds good!